File: python/1605. Find Valid Matrix Given Row and Column Sums.py
class Solution(object):
    def restoreMatrix(self, rowSum, colSum):
        """
        :type rowSum: List[int]
        :type colSum: List[int]
        :rtype: List[List[int]]
        """
        m, n = len(rowSum), len(colSum)
        grid = [[0]*n for _ in range(m)]
        for i in range(m):
            for j in range(n):
                now = min(rowSum[i], colSum[j])
                grid[i][j] = now
                rowSum[i] -= now
                colSum[j] -= now
        return grid
        

File: go/src/leetcode/1605/1605. Find Valid Matrix Given Row and Column Sums.go
package main

/*
*
https://leetcode.cn/problems/find-valid-matrix-given-row-and-column-sums/
*/
func restoreMatrix(rowSum []int, colSum []int) [][]int {
	m, n := len(rowSum), len(colSum)
	g := make([][]int, m)
	for i := range g {
		g[i] = make([]int, n)
	}
	for i := 0; i < m; i++ {
		for j := 0; j < n; j++ {
			g[i][j] = min(rowSum[i], colSum[j])
			rowSum[i] -= g[i][j]
			colSum[j] -= g[i][j]
		}
	}
	return g
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}
